subsys/bluetooth/mesh/prov_device.c

```diff
@@ -342,6 +344,15 @@ static void prov_dh_key_gen(void)
 	}
 }
 
+#if defined(CONFIG_BT_MESH_DHKEY_GEN_USE_BT_LONG_WQ)
+static void prov_dh_key_gen_handler(struct k_work *work)
+{
+	prov_dh_key_gen();
+}
+
+static K_WORK_DEFINE(dh_gen_work, prov_dh_key_gen_handler);
```

Comment:
What will happen if device/provisioner didn't succeed to generate dh pair but ack has been already sent?

Reply:
Ack acknowledges reception of the message but not the confirmation of generation of ECDH secret. If the device failes to generate ECDH secret, it will send another Provisioning Failed PDU: https://github.com/zephyrproject-rtos/zephyr/blob/9cec6db2fab528189aac1bd96ae560ba4c7f3fb3/subsys/bluetooth/mesh/provisioner.c#L435-L437